frontend/src/components/query-component/query-component.tsx

```diff
@@ -0,0 +1,34 @@
+import { IconButton, Input } from '@mui/material'
+import { ChangeEvent, useState } from 'react'
+import './query-component.scss'
+import { Send } from '@mui/icons-material'
+
+type QueryComponentProps = {
+  sendQuery: (query: string) => void
+}
+export function QueryComponent({ sendQuery }: QueryComponentProps) {
+  const [query, setQuery] = useState('')
+  return (
+    <div className="query-container">
+      <Input
+        className="query-input"
+        value={query}
+        onChange={(event: ChangeEvent<HTMLInputElement>) => {
+          setQuery(event.target.value)
+        }}
+        placeholder="Describe the product"
+        disableUnderline={true}
+      ></Input>
+      <IconButton
+        onClick={() => {
+          sendQuery(query)
+          setQuery('')
+        }}
```

Comment:
Should abstract out the callbacks as hooks. Definition of a listener declared with a lambda expression is never refreshed, while definition of a listener with a hook is refreshed when necessary.

```ts
export function QueryComponent({ sendQuery }: QueryComponentProps) {
  const handleClick = useCallback(() => {
    sendQuery();
    setQuery('');
  }, [sendQuery]);

  // Other parts

  return (
    // ...
    <IconButton
      onClick={handleClick}
      className="query-send-icon-button"
      aria-label="send"
    />
    // ...
  );
}
```

Reply:
I agree I have abstracted to a memoized function appropriately